haystack/nodes/prompt/providers.py

```diff
@@ -47,6 +49,15 @@ def __init__(self, model_name_or_path: str, **kwargs):
 
         self.model_name_or_path = model_name_or_path
 
+    def __init_subclass__(cls, **kwargs):
+        """
+        Used to register user-defined invocation layers.
+
+        Called when a subclass of PromptModelInvocationLayer is imported.
+        """
+        super().__init_subclass__(**kwargs)
+        cls.invocation_layer_providers.append(cls)
```

Comment:
This is the function that automatically registers all subclasses into the class attribute `invocation_layer_providers`